corehq/apps/programs/fixtures.py

```diff
@@ -43,13 +43,37 @@ def __call__(self, restore_state):
     def _get_fixture_items(self, restore_state):
         restore_user = restore_state.restore_user
 
-        def get_programs():
-            return Program.by_domain(restore_user.domain)
+        project = restore_user.project
+        if not project or not project.commtrack_enabled:
+            return []
+
+        data = Program.by_domain(restore_user.domain)
+
+        if not self._should_sync(data, restore_state.last_sync_log):
+            return []
 
         return simple_fixture_generator(
             restore_user, self.id, "program",
-            PROGRAM_FIELDS, get_programs, restore_state.last_sync_log
+            PROGRAM_FIELDS, data
         )
 
+    def _should_sync(self, data, last_sync):
+        """
+        Determine if a data collection needs to be synced.
+        """
+
+        # definitely sync if we haven't synced before
+        if not last_sync or not last_sync.date:
+            return True
+
+        # check if any items have been modified since last sync
+        for data_item in data:
```

Comment:
how many entries would this be?
If this can be high,
are you considering making a couch view for this?

Reply:
Generally very view